### PR TITLE
Fix portals unmount bug

### PR DESCRIPTION
**Pull request by savelichalex**
I don't know how it even worked after all 😅

### CODE REVIEW

File: kit/layout/src/Portal.tsx [hunk 1]
```diff
@@ -35,12 +35,17 @@ function PortalConsumer({ forId, absoluteFill, manager, children }: PortalConsum
...
             manager.update(key.current, children, forId, absoluteFill);
         }
+    }, [manager, children, forId, absoluteFill]);
 
+    React.useState(() => {
         return () => {
+            if (key.current == null) {
+                return;
+            }
             // @ts-ignore
```
Comment (sertony) [inline kit/layout/src/Portal.tsx]: Do we still need it?

File: kit/layout/src/Portal.tsx [hunk 2]
```diff
@@ -35,12 +35,17 @@ function PortalConsumer({ forId, absoluteFill, manager, children }: PortalConsum
...
-            manager.unmount(key.current, forId, absoluteFill);
+            manager.unmount(key.current, forId);
         };
-    }, [manager, children, forId, absoluteFill]);
+    });
```
Comment (sertony) [inline kit/layout/src/Portal.tsx]: I bet you've forgotten an empty array

```suggestion
    }, []);
```

File: kit/layout/src/Portal.tsx [hunk 3]
```diff
@@ -203,18 +203,19 @@ export class PortalManager extends React.PureComponent<PortalManagerProps, Porta
         const portals = this.state;
         return (
             Object.keys(portals)
-                .filter(portalKey => portals[Number(portalKey)] != null)
-                .sort((a, b) => Number(b) - Number(a)) // reversed order by keys
+                .map(portalKey => Number(portalKey))
```
Comment (sertony) [inline kit/layout/src/Portal.tsx:205]: Extra O(n) pass?)) OK, at least it's cleaner now.

File: kit/popups/src/Sheets/UISheet/UISheet.tsx [hunk 4]
```diff
@@ -144,6 +144,11 @@ function SheetContent({
             return undefined;
         }
 
+        if (!hasOpenAnimation) {
+            animate(true);
```
Comment (sertony) [inline kit/popups/src/Sheets/UISheet/UISheet.tsx]: if have no animation... then animate with `true`... 🤔 am I missing smth?

File: kit/layout/src/Portal.tsx (hunk 3 above)
Comment (savelichalex) [reply kit/layout/src/Portal.tsx:205]: And returns correct type (was`string` before that led to indexes - '1', '2', '21' etc)

File: kit/popups/src/Sheets/UISheet/UISheet.tsx (hunk 4 above)
Comment (savelichalex) [reply kit/popups/src/Sheets/UISheet/UISheet.tsx]: It's actually mean `open`)

**Review by SAnatoliiS — APPROVED** [review]

**Review by sertony — APPROVED** [review]

**Pull request merged**